homeassistant/components/freebox/binary_sensor.py

```diff
@@ -124,14 +124,12 @@ def __init__(
         self, hass: HomeAssistant, router: FreeboxRouter, node: dict[str, Any]
     ) -> None:
         """Initialize a cover for another device."""
-        cover_node = next(
-            filter(
-                lambda x: (x["name"] == self._sensor_name and x["ep_type"] == "signal"),
-                node["type"]["endpoints"],
-            ),
-            None,
-        )
-        super().__init__(hass, router, node, cover_node)
+        for endpoint in node["type"]["endpoints"]:
```

Comment:
Why not keep using `next` but replace the filter + lambda with a generator expression?